tso-server/src/main/java/org/apache/omid/tso/ReplyProcessorImpl.java

```diff
@@ -180,9 +192,19 @@ public void manageResponsesBatch(long batchSequence, Batch batch) {
 
     }
 
-    @Override
-    public void sendCommitResponse(long startTimestamp, long commitTimestamp, Channel c, MonitoringContext monCtx) {
+    @VisibleForTesting
+    void updateLowWatermark(Optional<Long> newLowwatermark) {
+        //TODO YONIGO - validate this is single threaded
```

Comment:
Have you? 😄 It should be, this processor does not use worker pool and it is single threaded.